src/Chainweb/Pact/PactService.hs

```diff
@@ -1449,6 +1542,8 @@ execPreInsertCheckReq txs = do
           cid = _chainId psEnv
       liftIO $ fmap Discard $ V.zipWith (>>)
         <$> validateChainwebTxs v cid cp parentTime lenientCreationTime currHeight txs (runGas pdb psState psEnv)
+            -- FIXME: it seems suspicious that the checkpointer is provided to the inner action.
```

Comment:
This is because `validateChainwebTxs` runs in `IO` due to the requirements of the mempool API, so `PactServiceM` is not available. 

The requirement is to do a dupe check https://github.com/kadena-io/chainweb-node/blob/3a5a5a73a711e22e61bfb3670ad9d6e3b50f0516/src/Chainweb/Pact/PactService.hs#L846-L851

note that `runGas` is a closure provided for the gas check and has the `pdb`/`psState`/`psEnv` for the current checkpointer. 

I would remove this FIXME (the real problem is the mempool API).

Reply:
oh, yes, that FIXME shouldn't be there. I just forgot to delete it.